### ISSUE TITLE
Add @theme variable to expressions

### ISSUE THREAD

**Issue by blacktesta** [qgis/QGIS#55427]
### Feature description

Hi, developers and community!
I ran into one problem. I wanted to filter my layers according to current map theme in QGIS and realised there is no way to do it. Provider feature filter is not a part of style, so it can't be changed by theme. Rule-based symbology can't get current theme because there is no @theme variable. That why i want to propose adding @theme or @currrent_theme variable in QGIS

### Additional context

P.s. making filter part of style would also be great

**Comment by blacktesta** [comment]
@DelazJ I want  
I want possibility to do something like this:
![image](https://github.com/qgis/QGIS/assets/74047515/89178b6f-fd59-4a37-82b4-0479cec6a01c)
Or possibility to include filter to style (like save to .qml)  so it can be changed with changing map theme.
So I'm talking not in context of layouts. I'm talking about main map in QGIS.

**Comment by szooon1** [comment]
@DelazJ Hi, 
I wanted to propose a feature like this, I think it would be possible to save a map theme with filters enabled on layers, not necessarily on all layers.  
I provide a link to a topic where I found helpful content: [https://gis.stackexchange.com/questions/470982/changing-layer-filter-according-to-map-theme-in-qgis](url)

**Comment by Sekou225** [comment]
> @DelazJ Hi, I wanted to propose a feature like this, I think it would be possible to save a map theme with filters enabled on layers, not necessarily on all layers. I provide a link to a topic where I found helpful content: [https://gis.stackexchange.com/questions/470982/changing-layer-filter-according-to-map-theme-in-qgis](url)

Dead link

**Comment by blacktesta** [comment]
Hi @Sekou225 . Thank you for your comment. The case you provided is not what i'm asking for, but i agree it would be great to have theme variable in expression builder . Attribute Form settings is already part of Layer Style, wich is part of Map Theme. So you can do the trick already just setting different forms in different styles in different themes.
I think I created a slight misunderstanding because I could not fully formulate what I wanted at the time of opening this issue.
Let say we have a layer with 200 features and one field "theme" with two possible values 'A' and 'B'. 100 of the features have value 'A'  and other 100 have value 'B'. And we have two map themes 'A' and 'B'. I want to filter this layer to have only 100 features with value 'A' when current theme is 'A'. At the moment i can do it with rule-based symbology, filtering features with rules. But in this case my users will see another 100 features with value 'B' in attribute table. So I need to use data provider filter, but I can't.
To sum up, I want to:
1. (complex solution) Have @theme variable and ability to use variables in provider feature filter query builder.
2. (less complex solution) Have ability to save provider feature filter query as part of layer style.

**Comment by blacktesta** [comment]
I have changed the title to better match the problem

**Comment by FrancoisAlto** [comment]
Hi everyone, 
Having a map theme variable in expression builder or a filter based on the map theme would be great. In my example, there is about 30 map themes in my project which call 30 different layer styles with  : 
- some symbology rules depending on a theme field value (to mask or to show features) 
- forms (which are similar in every layer style, in my case) 

to make my project easier to use and to set, i'd like to have only one layer style with symbology ruled by a variable like "@map_theme". When a user  chose a theme, the features with the same theme name  in the field "theme" would be showed.  
I know that multiple layer style or multiple symbologies rules would work but the problem is this one : 
When you have 30 layer styles and you realise there is an error in a form or in the symbology or a new field to add in the form, you have to repair the error 30 times so it needs work and there is a risk of forgetting some layer style. We have to do it manually because the style copy/paste does not work every time. (I don't know why)

If you have any other tips to help me i would be very grateful 
Thank you
